### PR TITLE
librbd: send_copyup() fixes and cleanups

### PR DESCRIPTION
**Pull request by idryomov**

### CODE REVIEW

**Comment by idryomov** [comment]
Updated to use ASSERT_PASSED so that the entire test aborts instead of just the helper function.

File: src/test/librbd/test_librbd.cc [hunk 1]
```diff
@@ -4477,6 +4477,67 @@ TEST_F(TestLibRBD, TestPendingAio)
...
+  librbd::Image clone_image;
+  ASSERT_EQ(0, rbd.open(ioctx, clone_image, clone_name.c_str(), NULL));
+  if (deep_copyup) {
+    ASSERT_EQ(0, clone_image.snap_create("snap1"));
+  }
+
+  bufferlist write_bl;
+  write_bl.append(std::string(512, '2'));
+  uint64_t mismatch_off;
+  ASSERT_EQ((ssize_t)write_bl.length(),
```
Comment (dillaman) [inline src/test/librbd/test_librbd.cc]: @idryomov Seeing a failure here when used against a real OSD [1]

> 2019-02-18T20:24:12.717 INFO:tasks.workunit.client.0.smithi185.stdout:      Expected: (ssize_t)write_bl.length()
> 2019-02-18T20:24:12.717 INFO:tasks.workunit.client.0.smithi185.stdout:      Which is: 512
> 2019-02-18T20:24:12.718 INFO:tasks.workunit.client.0.smithi185.stdout:To be equal to: clone_image.compare_and_write(512, write_bl.length(), bl, write_bl, &mismatch_off, 0)
> 2019-02-18T20:24:12.718 INFO:tasks.workunit.client.0.smithi185.stdout:      Which is: -84
> 

[1] http://qa-proxy.ceph.com/teuthology/jdillaman-2019-02-18_14:48:01-rbd-wip-jd-testing-distro-basic-smithi/3607746/teuthology.log

Comment (idryomov) [reply src/test/librbd/test_librbd.cc]: This is a bug in compare-and-write itself, working on a fix...

Comment (idryomov) [reply src/test/librbd/test_librbd.cc]: Opened https://github.com/ceph/ceph/pull/26519, will rebase on top of it once it's merged.

**Pull request merged**